backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/UserRepository.java

```diff
@@ -26,7 +29,23 @@
  * @author Johannes.Najjar@tngtech.com
  */
 
-@View(name = "all", map = "function(doc) { if (doc.type == 'user') emit(null, doc._id) }")
+@Views({
+        @View(name = "all",
+                map = "function(doc) { if (doc.type == 'user') emit(null, doc._id) }"),
+        @View(name = "byExternalId",
+                map = "function(doc) { if (doc.type == 'user') emit(doc.externalid, doc._id) }"),
+        @View(name = "byEmail",
+                map = "function(doc) { if (doc.type == 'user') emit(doc.email, doc._id) }"),
+        @View(name = "byFormerEmails",
```

Comment:
Why is there a `byFormerEmails` I would expect `byEmail` to contain also `byFormerEmails`.

Reply:
joined them as we discussed